### ISSUE TITLE
Better detection for GraphQL documents looks broken

### ISSUE THREAD

**Issue by FredyC** [dotansimha/graphql-code-generator#196]
I have query components defined like this. 

```tsx
import gql from 'graphql-tag'

import { buildQuery } from 'graph/buildQuery'
import { GPlayerList } from 'graph/types'

export const PlayerListQuery = buildQuery<
  GPlayerList.Query,
  GPlayerList.Variables
>(gql`
  query GPlayerList {
    allPlayers {
      id
      tier
      name
    }
  }
`)
```

While it was working with 0.8.18 just fine, the 0.8.19 ends with the error. Looks like there is quite deep dependency tree coming in with `strip-comments` and using `esprima` underneath which is unable to parse Typescript code. I would say that this change should be reverted and look for a more universal solution/module.

```
{ Error: Line 8: Unexpected token .
    at constructError (D:\workspace\app\scripts\node_modules\esprima\esprima.js:2407:21)
    at createError (D:\workspace\app\scripts\node_modules\esprima\esprima.js:2426:17)
    at unexpectedTokenError (D:\workspace\app\scripts\node_modules\esprima\esprima.js:2500:13)
    at throwUnexpectedToken (D:\workspace\app\scripts\node_modules\esprima\esprima.js:2505:15)
    at expect (D:\workspace\app\scripts\node_modules\esprima\esprima.js:2523:13)
    at parseLexicalBinding (D:\workspace\app\scripts\node_modules\esprima\esprima.js:4104:17)
    at parseBindingList (D:\workspace\app\scripts\node_modules\esprima\esprima.js:4120:23)
    at parseLexicalDeclaration (D:\workspace\app\scripts\node_modules\esprima\esprima.js:4181:24)
    at parseExportNamedDeclaration (D:\workspace\app\scripts\node_modules\esprima\esprima.js:5237:35)
    at parseExportDeclaration (D:\workspace\app\scripts\node_modules\esprima\esprima.js:5350:16)
    at parseStatementListItem (D:\workspace\app\scripts\node_modules\esprima\esprima.js:3970:24)
    at parseScriptBody (D:\workspace\app\scripts\node_modules\esprima\esprima.js:5490:25)
    at parseProgram (D:\workspace\app\scripts\node_modules\esprima\esprima.js:5506:16)
    at Object.parse (D:\workspace\app\scripts\node_modules\esprima\esprima.js:5690:23)
    at module.exports (D:\workspace\app\scripts\node_modules\esprima-extract-comments\index.js:26:21)
    at Comments.extract (D:\workspace\app\scripts\node_modules\extract-comments\lib\comments.js:52:11) lineNumber: 8, description: 'Unexpected token .', index: 197 }
```

**Comment by dotansimha** [comment]
Fixed in this commit: https://github.com/dotansimha/graphql-code-generator/commit/45d804565d2b8c5eda1c527498f4ff3c27c2e636  (just a fallback in case of esprima error)
I will release it soon. I added a test case for the specific file you mentioned. Any other issues?

**Comment by FredyC** [comment]
Ok thanks, that should do for now I guess, but in a situation you have comments in typescript file, it will essentially go back to original error :)

**Comment by dotansimha** [comment]
Sure, but I didn't want to hold you back with an old version :) 
I'll release it soon and i'll try to find a better solution later.

**Comment by dotansimha** [comment]
Fixed in 0.8.20 :)
